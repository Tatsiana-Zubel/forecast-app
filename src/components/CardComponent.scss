.card {
  border-style: solid dashed solid;
  padding: 10px;
  width: 160px;

  * {
    text-transform: uppercase;
  }
}