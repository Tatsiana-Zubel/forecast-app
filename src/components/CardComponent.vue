<template>
  <div class="card">
    <span class="title">{{ title }}</span>
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'CardComponent',
  props: {
    title: String
  }
}
</script>

<style lang="scss">
.card {
  border-style: solid dashed solid;
  padding: 10px;
  width: 160px;

  * {
    text-transform: uppercase;
    cursor: default;
  }
}
</style>
