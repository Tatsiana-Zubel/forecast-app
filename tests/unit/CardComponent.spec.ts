import { mount } from '@vue/test-utils'
import CardComponent from '@/components/CardComponent.vue'

describe('CardComponent.vue', () => {
  const wrapper = mount(CardComponent, {
    slots: {
      default: 'Main content'
    }
  })

  it('has "card" class', () => {
    const wrapperClass = wrapper.find('.card')
    expect(wrapperClass.exists())
  })

  it('has "title" class', () => {
    const wrapperClass = wrapper.find('.title')
    expect(wrapperClass.exists())
  })

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('checks slot', () => {
    const wrapperClass = wrapper.find('.card')
    expect(wrapperClass.text()).toContain('Main content')
  })

  it('sets prop title to "Test title"', async () => {
    await wrapper.setProps({ title: 'Test title' })
    expect(wrapper.vm.title).toBe('Test title')
  })
})
