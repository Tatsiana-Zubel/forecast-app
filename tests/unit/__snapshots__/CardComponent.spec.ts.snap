// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent.vue should match snapshot 1`] = `
<div
  class="card"
>
  <span
    class="title"
  />
  
  Main content
  
</div>
`;
